### ISSUE TITLE
Prevent GraphQL schema generation in discovery when license does not include GraphQL addon

### ISSUE THREAD

**Issue by day0ops** [k8sgateway/k8sgateway#7345]
### Is your feature request related to a problem? Please describe.

Prevent Discovery trying to generate GraphQL schemas with gRPC reflection when the license does not include GraphQL addon.

### Additional Context

Related to https://github.com/solo-io/gloo/issues/7343

**Comment by pseudonator** [comment]
I know that `discovery.fdsOptions.graphqlEnabled` has been introduced but that doesnt really solve the case where a user may not have GQL as an addon but GQL FDS will still get triggered right ? `discovery.fdsOptions.graphqlEnabled` is enabled ootb.

**Comment by kcbabo** [comment]
Yeah, I agree the UX is suboptimal there and should be fixed. I was just asking if using this setting for now is sufficient until we can add a license check to disable GQL discovery.
